.github/workflows/patch-release-build.yml

```diff
@@ -0,0 +1,121 @@
+# Releases a patch by cherrypicking commits into a release branch based on the previous
+# release tag.
+name: Patch Release Build
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: The version to tag the release with, e.g., 1.2.1, 1.2.2
+        required: true
+      commits:
+        description: Comma separated list of commit shas to cherrypick
+
+jobs:
+  prepare-release-branch:
+    runs-on: ubuntu-latest
+    outputs:
+      release-branch-name: ${{ steps.parse-release-branch.outputs.release-branch-name }}
+    steps:
+      - id: parse-release-branch
+        name: Parse release branch name
+        run: |
+          # Sets the release-branch-name output to the version number with the last non-period element replaced with an 'x' and preprended with v.
+          echo "::set-output name=release-branch-name::$(echo '${{ github.event.inputs.version }}' | sed -E 's/([^.]+)\.([^.]+)\.([^.]+)/v\1.\2.x/')"
+          # Sets the release-tag-name output to the version number with the last non-period element replace with a '0' and prepended with v
```

Comment:
Why is the tag of the patch version being replaced with 0 for the tag? Just want to make sure this workflow isn't actually pushing a new tag, it's just using this tag to checkout the minor version release that we're patching.

Reply:
Yup exactly, this is only to compute the tag to checkout to create the base of the release (hotfix) branch. The version will be something like `1.2.1` and we need to make sure we checkout `1.2.0` before cherrypicking and push that to `1.2.x`

Reply:
Makes sense. So we are still pushing a tag for the patch itself later right (L107)? One edge case is if we're doing a second patch release, say `1.2.2`, shouldn't we checkout the `1.2.1` tag not the `1.2.0` tag then? So shouldn't it compute `MAJOR.MINOR.(PATCH-1)`  as the tag to start from?

Reply:
After we've created the release branch, `1.2.x`, it's HEAD is always the previous patch release. An alternative would be to have branches for all of `1.2.1`, `1.2.2`, and then indeed we would check out from the previous tag - but it ends up with lots of branches so using `.x` is pretty conventional. It means we don't have to worry about checking out after the first patch release, we already have the release branch set up. It's why this is the `prepare-release-branch` job, it only runs once for the first patch (controlled by https://github.com/aws-observability/aws-otel-java-instrumentation/pull/16/files/e800882e7360e5b63ef23c24206528123a1c3526#diff-465a53fcd1e54767a6394a0fad650e8fd4872443de4dac07339011173c6c7480R34)

Reply:
Added comments :)

Reply:
Got ittttt so we only check out the minor release tag on the first patch release to make the `.x` branch then use that branch subsequently. Thanks for adding comments.